### PR TITLE
fix: tie-break top score by sorting by action name

### PR DESCRIPTION
**Pull request by typotter**
awesome set of tests triggering this bug [PR](https://github.com/Eppo-exp/sdk-test-data/pull/52)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -13,7 +13,7 @@ help: Makefile
 testDataDir := eppo-sdk-test/files
 tempDir := ${testDataDir}/temp
 gitDataDir := ${tempDir}/sdk-test-data
-branchName := main
+branchName := tp/fix/newflag
```
Comment (typotter) [inline Makefile:16]: temp override for test data fix

File: dot-net-sdk/validators/BanditEvaluator.cs [hunk 2]
```diff
@@ -105,7 +105,9 @@ public BanditEvaluator(int totalShards = 10_000)
     {
         var numberOfActions = actionScores.Count;
 
-        var bestAction = actionScores.Aggregate((currentmax, next) => currentmax.Value > next.Value ? currentmax : next);
+        // var bestAction = actionScores.Aggregate((currentmax, next) => currentmax.Value > next.Value ? currentmax : next);
+        // Order by key then by value to get the highest score, tie broken by action key.
+        var bestAction = actionScores.ToList().OrderByDescending(action=>action.Value).ThenBy(action=>action.Key).First();
```
Comment (typotter) [inline dot-net-sdk/validators/BanditEvaluator.cs:116]: What's the magnitude of the list of actions? The double sort is not the most efficient but it reads so nicely. Aggregating is O(1) but not as pretty to look at and understand.

File: Makefile (hunk 1 above)
Comment (aarsilv) [reply Makefile:16]: you can reset this now!

File: dot-net-sdk/validators/BanditEvaluator.cs (hunk 2 above)
Comment (aarsilv) [reply dot-net-sdk/validators/BanditEvaluator.cs:116]: I imagine most lists of actions will be 10-100.

Note that ChatGPT says `OrderByDescending()` chained with `ThenBy()` will still result in a single-pass sort

File: dot-net-sdk/validators/BanditEvaluator.cs [hunk 3]
```diff
@@ -105,7 +105,15 @@ public BanditEvaluator(int totalShards = 10_000)
...
         var numberOfActions = actionScores.Count;
 
-        var bestAction = actionScores.Aggregate((currentmax, next) => currentmax.Value > next.Value ? currentmax : next);
+        // var bestAction = actionScores.Aggregate((currentmax, next) =>
+        // {
+        //     if (currentmax.Value == next.Value) {
+        //         return currentmax.Key.CompareTo(next.Key) < 0 ? currentmax : next;
+        //     }
+        //     return currentmax.Value > next.Value ? currentmax : next;
+        // });
```
Comment (aarsilv) [inline dot-net-sdk/validators/BanditEvaluator.cs:114]: I'm fine axing this aggregate biasing towards the readability of below. Practically, I don't imagine it making much difference.

**Review by aarsilv — APPROVED** [review]

File: dot-net-sdk/validators/BanditEvaluator.cs (hunk 2 above)
Comment (typotter) [reply dot-net-sdk/validators/BanditEvaluator.cs:116]: I suspected that may be the case. thanks!

**Pull request merged**